src/middlewared/middlewared/plugins/pool.py

```diff
@@ -4069,5 +4085,51 @@ def parse_lsof(lsof, dirs):
     return list(pids.items())
 
 
+def parse_fstat(fstat, dirs):
+
+    # fstat output is separated by newlines
+    fstat = fstat.split('\n')
+
+    # first line is the column headers
+    fstat.pop(0)
+
+    # comprehension does a few things
+    #    1. remove any empty lines
+    #    2. remove any lines that dont have
+    #        at least 5 columns. The 5th
+    #        column is the path associated
+    #        with the process.
+    #    3. remove the g_eli,g_mirror information
+    #        from the output as its not needed
+    #        and ends up shifting those lines
+    #        quite a bit since the thread names
+    #        have spaces in their names
+    #    4. remove any lines that don't have a traditional
+    #        "path" since the path component for processes
+    #        can be a stream/socket/pipe etc
+    fstat = [
+        i for i in fstat if i and len(i.split()) >= 5 and not i.split()[1].startswith(('g_eli', 'g_mirror')) and '/' in i.split()[4]
+    ]
```

Comment:
```suggestion
    def keep(fields):
        if len(fields) < 5:
            return False
        if fields[1].startswith(('g_eli', 'g_mirror')):
            return False
        return '/' in fields[4]
    fstat = [ fields for fields in map(str.split, fstat) if keep(fields) ]
```

Reply:
This avoids splitting each line four times and IMO is easier to read.

Reply:
Good idea 👍